include/osmium/builder/osm_object_builder.hpp

```diff
@@ -324,7 +324,7 @@ namespace osmium {
 
         class ChangesetDiscussionBuilder : public Builder {
 
-            osmium::ChangesetComment* m_comment = nullptr;
+            std::size_t m_comment_offset = SIZE_MAX;
```

Comment:
This should use `std::numeric_limits<std::size_t>::max()` instead of `SIZE_MAX`. Same below.